client/scala/src/main/scala/ai/verta/repository/Commit.scala

```diff
@@ -23,6 +28,99 @@ class Commit(
     case _ => false
   }
 
+  /** Retrieves the blob at path from this commit
+   *  @param path location of a blob
+   *  @return ModelDB versioning blob. If not existed, return None
+   */
+  def get(path: String)(implicit ec: ExecutionContext): Try[Blob] = {
+    loadBlobs().flatMap(_ =>
+      blobs.get(path) match {
+        case None => Failure(new NoSuchElementException("No blob was stored at this path."))
+        case Some(blob) => Success(versioningBlobToBlob(blob))
+      }
+    )
+  }
+
+  /** Adds blob to this commit at path
+   *  If path is already in this Commit, it will be updated to the new blob
+   *  @param path Location to add blob to
+   *  @param blob Instance of Blob subclass.
+   *  @return whether the update attempt succeeds.
+   */
+  def update[T <: Blob](path: String, blob: T)(implicit ec: ExecutionContext): Try[Unit] = {
```

Comment:
How hard would it be to return the modified commit? I know the python client modifies in place, but different languages have different coding patterns.

Reply:
Hmm OK. How about return a whole new commit for every update/remove operation? I'll change the blobs map to be immutable, so that each such operation return a new version of it, but keep it a private var so that it's easier for initializing. How does that sound?

get and stuff doesn't modify or anything so we can disregard it.

Reply:
Sounds good